airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/context/env/ConnectorConfigurationPropertySource.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.context.env
+
+import com.fasterxml.jackson.core.type.TypeReference
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.commons.json.Jsons
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.env.MapPropertySource
+import io.micronaut.core.cli.CommandLine
+import io.micronaut.core.util.StringUtils
+import java.nio.file.Path
+import java.util.AbstractMap
+import java.util.stream.Collectors
+import java.util.stream.Stream
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * Custom Micronaut {@link PropertySource} that reads the command line arguments provided via the
+ * connector CLI and turns them into configuration properties. This allows the arguments to be
+ * injected into code that depends on them via Micronaut. <p /> This property source adds the
+ * following properties to the configuration if the matching value is present in the CLI arguments:
+ * <ul> <li><b>airbyte.connector.operation</b> - the operation argument (e.g. "check", "discover",
+ * etc)</li> <li><b>airbyte.connector.catalog</b> - the Airbyte configured catalog as JSON read from
+ * the configured catalog file path argument, if present.</li> <li><b>airbyte.connector.config</b> -
+ * the normalized Airbyte connector configuration read from the configured connector configuration
+ * file path argument, if present.</li> <li><b>airbyte.connector.state</b> - the Airbyte connector
+ * state as JSON read from the state file path argument, if present.</li> </ol>
+ */
+class ConnectorConfigurationPropertySource(commandLine: CommandLine) :
+    MapPropertySource("connector", resolveValues(commandLine)) {
+    companion object {
+        private const val PREFIX_FORMAT = "%s.%s"
```

Comment:
@postamar it is used to separate the config, state and catalog into separate namespaces to avoid collision.  All of these end up under the `airbyte` namespace with the prefix with the type being the first parameter of the prefix and the name of the object as the second half.  This is necessary for the mapping on the actual objects works, as you have to provide a root path on the object and then Micronaut uses the names of the fields in the objects to map things under that root path.